pkg/controller/controller.go

```diff
@@ -154,8 +156,14 @@ func (b *controller) worker(ctx context.Context) {
 			// Increase sync count for this controller
 			b.metrics.IncrementSyncCallCount(b.name)
 
-			if err := b.syncHandler(ctx, key); err != nil {
-				log.Error(err, "re-queuing item due to error processing")
+			err := b.syncHandler(ctx, key)
+			if err != nil {
+				if strings.Contains(err.Error(), genericregistry.OptimisticLockErrorMsg) {
+					log.V(logf.DebugLevel).Error(err, "re-queuing item due to optimistic locking")
```

Comment:
Knowing when a resync happens due to optimistic locking is still important information to anyone since it allows to understand that the sync is being retried (useful to understand why the same log line appears twice for example).

Pushing this log line to `-v=4` seems unfortunate to me since most users use `-v=2` at this point. Would it make sense to use logf.InfoLevel + `Info` instead?
```suggestion
					log.V(logf.InfoLevel).Info(err, "re-queuing item due to optimistic locking")
```

Reply:
^ what do you think?